File: tests/vb/vb_comprehensive/nested_for_loops.vb
' vybe-test: vb/vb_comprehensive/nested_for_loops
' origin: languages/vb/tests/vb/vb_comprehensive_test.rs

Module M
    Sub Main()
        Dim total As Integer = 0
        For i As Integer = 1 To 3
            For j As Integer = 1 To 3
                total = total + 1
            Next
        Next
        Console.WriteLine(total)
    End Sub
End Module

File: tests/csharp/control_flow/nested_for_loops.cs
// vybe-test: csharp/control_flow/nested_for_loops
// origin: languages/csharp/tests/csharp/test_control_flow.rs

using static __Harness;

var sum = 0;
for (var i = 0; i < 3; i++) {
            for (var j = 0; j < 3; j++) {
                sum = sum + 1;
            }
        }
__P((sum).ToString());
__Check("9");

public static class __Harness {
    public static string __buf = "";
    public static void __P(string s) { __buf = __buf + s + "\n"; }
    public static void __Pr(string s) { __buf = __buf + s; }
    public static void __Check(string want) {
        if (__buf != want && __buf != want + "\n") {
            Console.WriteLine("FAIL: want [" + want + "] got [" + __buf + "]");
            throw new Exception("assertion failed");
        }
    }
}
